lib/airbrake-ruby/code_hunk.rb

```diff
@@ -12,8 +12,9 @@ class CodeHunk
     # @return [Integer] how many lines should be read around the base line
     NLINES = 2
 
-    def initialize(config)
+    def initialize(config, cache)
       @config = config
+      @cache = cache
```

Comment:
Can you create cache here instead of passing it as arg?

Reply:
It will be useless because it will be per frame, not per backtrace.

Reply:
Can we make CodeHunk and cache instances global?

Reply:
Not sure what exactly you suggest, but that doesn't sound thread-safe friendly.

Reply:
I suggest that there is exactly one instance of `CodeHunk` in the notifier. And all code hunks are cached in single cache instance. Thread-safety can be achieved with mutexes where needed.